src/catch2/matchers/catch_matchers_exception.hpp

```diff
@@ -29,6 +29,32 @@ class ExceptionMessageMatcher final : public MatcherBase<std::exception> {
 //! Creates a matcher that checks whether a std derived exception has the provided message
 ExceptionMessageMatcher Message(std::string const& message);
 
+template <typename StringMatcherType>
```

Comment:
Do you intend for this to take arbitrary matcher, rather than just the ones derived from `StringMatcherBase`?

Reply:
The regex matcher might be useful and it seems not to inherit from `StringMatcherBase`
Do we gain anything if we limit this to a subclass of `StringMatcherBase`?

Reply:
Ok, makes sense.